### ISSUE TITLE
http pipelining not working?

### ISSUE THREAD

**Issue by baslr** [arangodb/arangodb#1258]
Hi,

I wrote a short test to try http pipelining:

This will send 2 GET requests in a row without waiting for the response.
```coffeescript
net = require 'net'

socket = net.connect 8000, '127.0.0.1'

socket.on 'data', (data) -> console.dir data.toString 'utf8'

socket.on 'connect', () ->
    str = ''
    for i in [0..1]
      str += "GET /_db/_system/_api/document/test/#{i} HTTP/1.1\r\n\r\n"

    console.dir str
    console.log()

    socket.write str, 'utf8'
```
```
# two GETs
'GET /_db/_system/_api/document/test/0 HTTP/1.1\r\n\r\nGET /_db/_system/_api/document/test/1 HTTP/1.1\r\n\r\n'

# only one response
'HTTP/1.1 404 Not Found\r\nServer: ArangoDB\r\nConnection: Keep-Alive\r\nContent-Type: application/json; charset=utf-8\r\nContent-Length: 103\r\n\r\n{"error":true,"errorMessage":"document \\/_api\\/document\\/test\\/0 not found","code":404,"errorNum":1202}
```
But I only get one response. Something wrong with my test? Issue #774 does not have an example.
Server is arango 2.5.0-beta2 on OS X

Thanks for your help.

**Comment by jsteemann** [comment]
Your test code looks good. 
We found an issue on our end. The pipelining did work for some types of requests (the ones that are executed directly) but not for others (the ones that are put onto an async task queue inside ArangoDB). It should be fixed with the above commits.
I'll leave this issue open though because I still want to add further tests before closing.

**Comment by jsteemann** [comment]
Tests for both direct and indirect request handlers have been added. 
Issue should be fixed in 2.4, 2.5 and devel branches and shipped with next releases.

**Comment by weinberger** [comment]
Ich leider auch nicht.
